draft-ietf-httpbis-http2bis.xml

```diff
@@ -606,6 +600,42 @@ HTTP Frame {
           connection with a <xref target="ConnectionErrorHandler">connection error</xref> of type
           <xref target="COMPRESSION_ERROR" format="none">COMPRESSION_ERROR</xref> if it does not decompress a field block.
         </t>
+        <t>
+          A decoding error in a field block MUST be treated as a <xref
+          target="ConnectionErrorHandler">connection error</xref> of type <xref
+          target="COMPRESSION_ERROR" format="none">COMPRESSION_ERROR</xref>.
+        </t>
+
+        <section>
+          <name>Compression State</name>
+          <t>
+            Field compression is stateful.  One compression context and one decompression context
+            are used for the entire connection.  <xref target="COMPRESSION" section="4"/> defines
+            the dynamic table, which is the primary state that is used for field compression.
+          </t>
+          <t>
+            The dynamic table has a maximum size that is set by a decoder using the
+            SETTINGS_HEADER_TABLE_SIZE setting; see <xref target="SettingValues"/>.  The encoder can
+            set the dynamic table to any size up to the maximum value set by the decoder.  The
+            encoder declares the size of the dynamic table with a Dynamic Table Size Update
+            instruction (<xref target="COMPRESSION" section="6.3"/>).  The encoder at both client
+            and server is initialized with a dynamic table size of 4,096 bytes, the initial value of
+            the SETTINGS_HEADER_TABLE_SIZE setting.
+          </t>
+          <t>
+            Any change to the maximum value set by the decoder takes effect when the encoder <xref
+            target="SettingsSync">acknowledges settings</xref>.  The first field block sent by an
+            encoder after a change in SETTINGS_HEADER_TABLE_SIZE starts with at least one Dynamic
+            Table Size Update instruction; see <xref target="COMPRESSION" section="4.2"/>.  An
+            encoder sends a Dynamic Table Size Update instruction after acknowledging a change of
+            SETTINGS_HEADER_TABLE_SIZE even if it is not changing the size of the dynamic table or
```

Comment:
Is this correct?

Quoting from [RFC 7541 section 4.2](https://httpwg.org/specs/rfc7541.html#maximum.table.size): _A change in the maximum size of the dynamic table is signaled via a dynamic table size update (see Section 6.3). This dynamic table size update MUST occur at the beginning of the first header block following the change to the dynamic table size. In HTTP/2, this follows a settings acknowledgment (see Section 6.5.3 of [HTTP2])._

This paragraph of RFC 7541 starts "a change," but if the an HTTP/2 using an HPACK encoder with a dynamic table of size 4,096 (the default) and receives SETTINGS_MAX_HEADER_LIST_SIZE of 4096, it is not changing the size of the dynamic table.

Interoperability-wise, do existing servers respond with a Dynamic Table Update Size instruction when the client sends SETTINGS_MAX_HEADER_LIST_SIZE greater than 4096? It seems to me that this newly added text seem to suggest such behavior is required, but I'm not sure if many servers behave as such.